### DISCUSSION TITLE
Proposal: tvg-id should match what exists in iptv-org/epg

### DISCUSSION THREAD

**Discussion by 13steinj** [iptv-org/community#246] [Ideas]
Rationale:

tvg-id is a non-standard identifier in m3u playlists. Technically, any provider (m3u / epg) can change them at any time.

However, I would hope that when one org / person (thank you for the work by the way) is providing *both*, that the two are internally consistent with each other (with the iptv-org/epg 's `xmltv_id`).

Many mappers can't automatically map the m3u files to the epg data as a result. When there's many channels (on the order of several dozen), it get's *very* annoying to remap this manually with some basic heuristic, as most mapper's don't have an API (personally, mine does, but I know that many other non-technical ones do not have one) and the UI is traditionally beyond cumbersome in that respect (mass manual mapping), if it even exists at all.

Details:

Just for example, most in the epg repo (for the US, I don't know about the others yet) use the station's call sign as the `xmltv_id`, if the call sign also exists for radio, then specification of "TV" at the end, sometimes with an additional number (some don't use a call sign at all, don't know why).

I would request (and I can do this if you'd like), that for channels listed in the epg repo, the call signs are parsed out. Using regex or otherwise, filter out the "TV", so all you get is the call sign and the number, then search for that call sign and number in the iptv repo's m3u. If found, replace the tvg id (subject to review).

Preferably the channel names should be consistent as well.

Note: I do not know if the logo in channels for the epg repo is correct or not, it appears to use the generic logo, whereas the m3u files use the station / region specific logo. Perhaps this should also be standardized across the two repos (and using the more specific logo).

E: To clarify, I believe this is different than #2438 because (unless I'm misunderstanding), the contributor's response was to provide a program to map this externally, whereas this should be done internally (because again, almost every mapping program is cumbersome to work with at best). The difference in logos or names may also be relevant.

**Comment by BellezaEmporium** [comment]
Right now, it's being in study, and @freearhey 's the one behind the idea. He wants everything done correctly in iptv-org/epg before merging and adding the EPGs into iptv-org/iptv.

It is only a matter of time now, stay tuned for more information linked to the association.
